toolchain/sem_ir/typed_insts.h

```diff
@@ -1948,6 +1948,14 @@ struct WitnessType {
   TypeId type_id;
 };
 
+// HasInstCategory is true if T::Kind is an element of AnyInstT::Kinds.
+template <typename AnyInstT, typename T>
+concept HasInstCategory = requires {
+  T::Kind;
+  requires std::find(std::begin(AnyInstT::Kinds), std::end(AnyInstT::Kinds),
+                     T::Kind) != std::end(AnyInstT::Kinds);
```

Comment:
This is reimplementing functionality that's already available as `InstLikeTypeInfo<T>::IsKind<AnyInstT>()`; can you refactor to avoid the duplication?

(Also, can we use `std::contains` yet?)

Reply:
> This is reimplementing functionality that's already available as `InstLikeTypeInfo<T>::IsKind<AnyInstT>()`; can you refactor to avoid the duplication?

Oh, cool, I missed that. Done, and I took the opportunity to simplify the definition while I was at it.

> (Also, can we use `std::contains` yet?)

AFAICT there is no `std::contains`, only `std::ranges::contains`, and we don't use `std::ranges`.